trieste/acquisition/rule.py

```diff
@@ -1050,7 +1112,7 @@ def select_in_region(self, mapping: Optional[Mapping[Tag, T]]) -> Optional[Mappi
         elif self.region_index is None:
             # If no index, then return the global items.
             _mapping = {
-                tag: item
+                tag: self.with_input_active_dims(item)
```

Comment:
slightly surprised this type checks, given T isn't guaranteed to satisfy the input requirements for with_input_active_dims. Can you bind T to `Union[TensorType, Dataset, ProbabilisticModel]`?

Reply:
Good spot. Not sure why typing works. I have changed it.